src/main/webapp/app/shared/components/code-button/code-button.component.ts

```diff
@@ -79,27 +79,22 @@ export class CodeButtonComponent implements OnInit, OnChanges {
         private ideSettingsService: IdeSettingsService,
     ) {}
 
-    ngOnInit() {
-        this.accountService
-            .identity()
-            .then((user) => {
-                this.user = user!;
-                this.refreshTokenState();
-
-                this.copyEnabled = true;
-                this.useSsh = this.localStorage.retrieve('useSsh') || false;
-                this.useToken = this.localStorage.retrieve('useToken') || false;
-                this.localStorage.observe('useSsh').subscribe((useSsh) => (this.useSsh = useSsh || false));
-                this.localStorage.observe('useToken').subscribe((useToken) => (this.useToken = useToken || false));
-
-                if (this.useSsh) {
-                    this.useSshUrl();
-                }
-                if (this.useToken) {
-                    this.useHttpsUrlWithToken();
-                }
-            })
-            .then(() => this.loadParticipationVcsAccessTokens());
+    async ngOnInit() {
+        this.user = (await this.accountService.identity())!;
```

Comment:
Isn't it a bit dangerous to assume that the accountService always returns a valid user?

Reply:
I changed it, but tbh. I'm not sure if that really improves the situation. The user should always be defined here, but if it's not, I'm really unsure on how that would need to be handled, to not leave the code button in a broken state. (the user is basically needed for everything)

Reply:
Yeah fair :)